src/components/PopupWithForm.js

```diff
@@ -26,4 +34,26 @@ export default class PopupWithForm extends Popup {
     });
     super.setEventListeners();
   }
+  //***SET MODAL INPUTS***
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name];
+    });
+  }
+  //***CHANGE BUTTON TEXT WITH RENDER LOADING/ TRIED THE SUGGESTIONS BUT I RECEIVED SOME ERRORS***
+  /*renderLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButtonText = this._button.textContent;
+      this._formSubmit = loadingText;
+    } else {
+      this._button.textContent = this._submitButtonText;
+    }
+  }
```

Comment:
This is the correct variant of `renderLoading` in your project if you want to implement it

```js

  renderLoading(isLoading, loadingText = "Saving...") {
    if (isLoading) {
      this._button.textContent = loadingText
    } else {
      this._button.textContent = this._originalButtonText;
    }
  }

```